index.js

```diff
@@ -96,47 +105,34 @@ class QuickLRU {
 
 	set(key, value) {
 		if (this.cache.has(key)) {
-			if (this.maxAge > 0) {
-				this.cache.set(key, {
-					value,
-					expiry: Date.now() + this.maxAge
-				});
-			} else {
-				this.cache.set(key, value);
-			}
+			this._set(key, value, true);
 		} else {
-			this._set(key, value);
+			this._set(key, value, false);
 		}
 
 		return this;
 	}
 
 	has(key) {
 		if (this.maxAge > 0) {
-			return Boolean(this._getOrDeleteIfExpired(
+			const item = this.cache.get(key) || this.oldCache.get(key);
```

Comment:
This is incorrect.

From the package description:

> Inspired by the hashlru algorithm, but instead uses Map to support keys of any type, not just strings, **and values can be undefined.**

If the value is `undefined` here, it falls back to the `oldCache`, which is incorrect. Please add lots of tests for the `undefined` value case with the different methods when you fix this.

Reply:
Yep you are right, let me add more test for this cases ;)